tSubject: Changed tmpdir default to /var/tmp

By default, mutt stores drafts in
/tmp/mutt-$HOSTNAME-$UID-$PID-$REV. /tmp is designed to hold data
that isn't guaranteed to be preserved between different invocations
of programmes. For practical purposes, and on Debian at least, this
translates to files being deleted on every boot, unless TMPTIME is
amended in /etc/default/rcS and /tmp is not a tmpfs.

This behaviour can cause loss of data, e.g. when the laptop dies or
turns itself off due to low power; the user might not have a chance
to commit the temporary mail draft s/he was working on to
non-volatile storage.

I suggest that mutt should use /var/tmp by default for its mail
drafts, as that directory is also temporary in nature, but cleaned
less frequently than /tmp. More specifically, the FHS dictates that
"files and directories located in /var/tmp must not be deleted when
the system is booted."

This patch modifies all occurrences of $tmpdir, $TMPDIR and Tempdir, but
not the documentation, which looks like it's generated from init.h.

Patch created by: martin f. krafft <madduck@madduck.net>, submitted to
http://dev.mutt.org/trac/ticket/3236
and repaired to be able to apply.

Fedora related: Upstream is not willing to change defualt tmpdir,
because they don't believe /var/tmp is on all supported platforms.
However, they agree with changing it in distrubution packages. 
More info at http://dev.mutt.org/trac/ticket/3236

diff --git a/doc/manual.xml.head b/doc/manual.xml.head
index 03ab29c..56a6e37 100644
--- a/doc/manual.xml.head
+++ b/doc/manual.xml.head
@@ -6253,7 +6253,7 @@ Attachments appear as follows by default:
 </para>
 
 <screen>
-- 1 [text/plain, 7bit, 1K]           /tmp/mutt-euler-8082-0 &lt;no description&gt;
+- 1 [text/plain, 7bit, 1K]           /var/tmp/mutt-euler-8082-0 &lt;no description&gt;
   2 [applica/x-gunzip, base64, 422K] ~/src/mutt-0.85.tar.gz &lt;no description&gt;
 </screen>
 
diff --git a/doc/mutt.man b/doc/mutt.man
index 8a5113c..dd9fab4 100644
--- a/doc/mutt.man
+++ b/doc/mutt.man
@@ -173,7 +173,7 @@ Specifies the editor to use when composing messages.
 User configuration file.
 .IP "@sysconfdir@/Muttrc"
 System-wide configuration file.
-.IP "/tmp/muttXXXXXX"
+.IP "/var/tmp/muttXXXXXX"
 Temporary files created by Mutt.
 .IP "~/.mailcap"
 User definition for handling non-text MIME types.
diff --git a/init.c b/init.c
index a4ab3a0..ca28293 100644
--- a/init.c
+++ b/init.c
@@ -3002,7 +3002,7 @@ void mutt_init (int skip_sys_rc, LIST *commands)
     MailcapPath = safe_strdup ("~/.mailcap:" PKGDATADIR "/mailcap:" SYSCONFDIR "/mailcap:/etc/mailcap:/usr/etc/mailcap:/usr/local/etc/mailcap");
   }
 
-  Tempdir = safe_strdup ((p = getenv ("TMPDIR")) ? p : "/tmp");
+  Tempdir = safe_strdup ((p = getenv ("TMPDIR")) ? p : "/var/tmp");
 
   p = getenv ("VISUAL");
   if (!p)
diff --git a/init.h b/init.h
index 190e2af..36cbea9 100644
--- a/init.h
+++ b/init.h
@@ -3286,7 +3286,7 @@ struct option_t MuttVars[] = {
   ** This variable allows you to specify where Mutt will place its
   ** temporary files needed for displaying and composing messages.  If
   ** this variable is not set, the environment variable \fC$$$TMPDIR\fP is
-  ** used.  If \fC$$$TMPDIR\fP is not set then ``\fC/tmp\fP'' is used.
+  ** used.  If \fC$$$TMPDIR\fP is not set then ``\fC/var/tmp\fP'' is used.
   */
   { "to_chars",		DT_STR,	 R_BOTH, UL &Tochars, UL " +TCFL" },
   /*
diff --git a/muttbug.sh.in b/muttbug.sh.in
index b429fae..705e894 100644
--- a/muttbug.sh.in
+++ b/muttbug.sh.in
@@ -85,7 +85,7 @@ esac
 exec > /dev/tty
 exec < /dev/tty
 
-SCRATCH=${TMPDIR-/tmp}/`basename $0`.`hostname`.$$
+SCRATCH=${TMPDIR-/var/tmp}/`basename $0`.`hostname`.$$
 
 mkdir ${SCRATCH} || \
 { 
